dockerfiles/stages/4-production

```diff
@@ -59,8 +59,21 @@ RUN test "$deb_codename" = "focal" \
          libffi7 \
      || exit 0
 
-# --- Install the latest version of rosetta-cli
-RUN curl -sSfL https://raw.githubusercontent.com/coinbase/rosetta-cli/master/scripts/install.sh | sh -s
+# --- Install rosetta-cli
+
+# The following commands install rosetta-cli with an updated rosetta-sdk-go dependency, which supports delegation transactions.
+# Upstream PR to rosetta-sdk-go: https://github.com/coinbase/rosetta-sdk-go/pull/457
+# They can be replaced once there is a new release of rosetta-cli containing the above change by:
+# RUN curl -sSfL https://raw.githubusercontent.com/coinbase/rosetta-cli/${ROSETTA_CLI_VERSION}/scripts/install.sh | sh -s
+COPY --from=builder /usr/lib/go /usr/lib/go
+RUN apt-get install -y gcc
+RUN export GOBIN="$(pwd)/bin" \
+  && curl -L "https://github.com/coinbase/rosetta-cli/archive/refs/tags/v0.10.1.tar.gz" -o "/tmp/v0.10.1.tar.gz" \
+  && tar xzf "/tmp/v0.10.1.tar.gz" -C "/tmp" \
+  && cd "/tmp/rosetta-cli-0.10.1" \
+  && /usr/lib/go/bin/go mod edit -replace github.com/coinbase/rosetta-sdk-go@v0.8.1=github.com/ylecornec/rosetta-sdk-go@v0.0.0-20221213104605-e9d9f74909c0 \
```

Comment:
`github.com/ylecornec/rosetta-sdk-go@v0.0.0-20221213104605-e9d9f74909c0` is a personal repository. If this repository status changes or is deleted, it will break the build. We should better having a fork in the MinaProtocol Organization